components/scrubber/scrubber.go

```diff
@@ -86,6 +97,8 @@ type Scrubber interface {
 	//   }
 	//
 	Struct(val any) error
+
+	DeepCopyStruct(val any) any
```

Comment:
Could you add docs explaining the different to Struct? Does it create a new object while scrubbing? What are expectations about interfaces?